process-database/common-database-patterns.rst

```diff
@@ -54,5 +54,43 @@ In the following example the value read from a ‘bi’ is inverted so that the
         field(FLNK, "$(P):rbv")
     }
 
-The important point is the NPP modifier on output link of the ‘calcout’ record. 
+The important point is the NPP modifier on output link of the ‘calcout’ record.
 This updates the VAL field of the ‘bo’ record (and posts monitors) without processing it.
+
+Readback from a different hardware address
+------------------------------------------
+Use when the readback value of an ‘out‘ record comes from a different hardware address. In this case it can be undesirable for an update of the readback value to trigger processing of the ‘out‘ record again as this could lead to infinite loops. SDIS is used to ensure that writes to the hardware only happen when they are not triggered by the readback value changing. The main record is a soft record, this ensures that monitors are updated when its value changes.
+
+In the following example a longout record gets its readback value from a different longin record.
+
+::
+
+    record(longout, "$(P)") {
+        field(OUT,  "$(P):write PP")
+    }
+
+::
+
+    record(longout, "$(P):write") {
+        field(DTYP, "...")
+        field(OUT,  "...")
+        field(SDIS, "$(P):update.PACT")
+    }
+
+::
+
+    record(calcout, "$(P):update") {
+        field(CALC, "A")
+        field(INPA, "$(P):rbv")
+        field(OUT,  "$(P) PP")
+    }
```

Comment:
I understand that in some cases having the ability to perform a calculation here might be useful, but for this particular example I recommend replacing the calcout with a longout with `DOL = "$(P):rbv"`, `OMSL = "closed_loop"` and `OUT = "$(P) PP"`. The `calcout::process()` routine is quite complex, and using that involves converting the int32 from `$(P):rbv.VAL` to a double when reading `INPA`, calling the expression evaluation code to interpret an RPN expression that gets `A` and writes it into `VAL`, then converting `VAL` back to int32 when writing to the `$(P)` record. There are quite a few other checks which go on during `calcout::process()` to see if additional functionality has been configured, and they all read other field values and take a little bit of CPU time to do.